lib/inventory/inventory.go

```diff
@@ -71,7 +71,7 @@ type DownstreamHandle interface {
 	// any clean up of the connection. A Goodbye is merely information so that the
 	// upstream half of the connection may take different actions when the downstream
 	// half of the connection is shutting down for good vs. restarting.
-	SendGoodbye(context.Context) error
+	SendGoodbye(context.Context, bool) error
```

Comment:
nit: comments on this PR still specify that goodbyes are specifically sent by agents that are shutting down for good rather than restarting.  Comments here (and possibly elsewhere) should be amended to indicate that goodbyes are a general purpose message informing the control-plane about the nature of the agent's shutdown.